src/preamble.js

```diff
@@ -1017,9 +1019,9 @@ function enlargeMemory() {
 
   var LIMIT = Math.pow(2, 31); // 2GB is a practical maximum, as we use signed ints as pointers
                                // and JS engines seem unhappy to give us 2GB arrays currently
-  if (DYNAMICTOP >= LIMIT) return false;
+  if (HEAPU32[DYNAMICTOP_PTR>>2] >= LIMIT) return false;
```

Comment:
Updated these to be stricter. I now realize that 2GB exact is off limits, but 2BG - 16MB should be attainable.  There is some room for improvement in this heuristic so that we could realistically get that as a max cap, although that is not quite yet urgent.